lib/runtime/wasmer/imports_test.go

```diff
@@ -1492,3 +1494,93 @@ func Test_ext_trie_blake2_256_root_version_1(t *testing.T) {
 	expected := tt.MustHash()
 	require.Equal(t, expected[:], hash)
 }
+
+func Test_ext_trie_blake2_256_verify_proof_version_1(t *testing.T) {
+	t.Parallel()
+
+	tmp, err := ioutil.TempDir("", "*-test-trie")
+	require.NoError(t, err)
+
+	memdb, err := chaindb.NewBadgerDB(&chaindb.Config{
+		InMemory: true,
+		DataDir:  tmp,
+	})
+	require.NoError(t, err)
+
+	otherTrie := trie.NewEmptyTrie()
+	otherTrie.Put([]byte("simple"), []byte("cat"))
+
+	otherHash, err := otherTrie.Hash()
+	require.NoError(t, err)
+
+	tr := trie.NewEmptyTrie()
+	tr.Put([]byte("do"), []byte("verb"))
+	tr.Put([]byte("domain"), []byte("website"))
+	tr.Put([]byte("other"), []byte("random"))
+	tr.Put([]byte("otherwise"), []byte("randomstuff"))
+	tr.Put([]byte("cat"), []byte("another animal"))
+
+	err = tr.Store(memdb)
+	require.NoError(t, err)
+
+	hash, err := tr.Hash()
+	require.NoError(t, err)
+
+	keys := [][]byte{
+		[]byte("do"),
+		[]byte("domain"),
+		[]byte("other"),
+		[]byte("otherwise"),
+		[]byte("cat"),
+	}
+
+	root := hash.ToBytes()
+	otherRoot := otherHash.ToBytes()
+
+	proof, err := trie.GenerateProof(root, keys, memdb)
+	require.NoError(t, err)
+
+	testcases := []struct {
+		root, key, value []byte
+		proof            [][]byte
+		expect           bool
+	}{
+		{root: root, key: []byte("do"), proof: proof, value: []byte("verb"), expect: true},
+		{root: []byte{}, key: []byte("do"), proof: proof, value: []byte("verb"), expect: false},
+		{root: otherRoot, key: []byte("do"), proof: proof, value: []byte("verb"), expect: false},
+		{root: root, key: []byte("do"), proof: proof, value: nil, expect: true},
+		{root: root, key: []byte("unknow"), proof: proof, value: nil, expect: false},
+		{root: root, key: []byte("unknow"), proof: proof, value: []byte("unknow"), expect: false},
+		{root: root, key: []byte("do"), proof: [][]byte{}, value: nil, expect: false},
+	}
+
+	inst := NewTestInstance(t, runtime.HOST_API_TEST_RUNTIME)
+
+	for _, testcase := range testcases {
+		hashEnc, err := scale.Marshal(testcase.root)
```

Comment:
You can also run each test case in parallel with something like:

```suggestion
	for _, testcase := range testcases {
		testcase := testcase
		t.Run(testcase.name, func(t *testing.T) {
			t.Parallel()
			
			hashEnc, err := scale.Marshal(testcase.root)
```

Instead of adding a field `name` to testcase, you could also have your testcases as a `map[string]struct{...}` such that the key is the test case name.

Then you might want to run the test with the `-race` flag to make sure there is no data races, but that is optional I guess.

Reply:
Done!